lib/logic/task_instance/task_instance_cubit.dart

```diff
@@ -0,0 +1,159 @@
+		List<PlanInstance> toUpdate = [];
+		bool wasPlanInstanceChanged = false;
+
+		if(planInstance.state != PlanInstanceState.active) {
+			planInstance.state = PlanInstanceState.active;
+			wasPlanInstanceChanged = true;
+
+			var childId = _activeUser().id;
+			var allPlans = await _dataRepository.getPlans(childId: childId);
+			var todayPlans = await _repeatabilityService.filterPlansByDate(allPlans, Date.now());
+			var todayPlanIds = todayPlans.map((plan) => plan.id).toList();
+			var untilCompletedPlans = allPlans.where((plan) => plan.repeatability.untilCompleted && !todayPlans.contains(plan.id)).map((plan) => plan.id).toList();
+			var instances = await _dataRepository.getPlanInstancesForPlans(childId, todayPlanIds, Date.now());
+			instances.addAll(await _dataRepository.getPastNotCompletedPlanInstances([childId], untilCompletedPlans, Date.now()));
+
+			for(var instance in instances) {
+				if(instance.state == PlanInstanceState.active) {
+					instance.state = PlanInstanceState.notCompleted;
+					toUpdate.add(instance);
+					break;
+				}
+			}
+		}
+
+		if((taskInstance.duration.length == 0 || (taskInstance.duration.length > 0 && taskInstance.duration.last.to != null)) && (taskInstance.breaks.length == 0 || (taskInstance.duration.length > 0 && taskInstance.breaks.last.to != null))) {
+			if(taskInstance.duration == null) taskInstance.duration = [];
+			if(taskInstance.status.completed) taskInstance.status.completed = false;
+			taskInstance.duration.add(DateSpan(from: TimeDate.now()));
+			updates.add(_dataRepository.updateTaskInstance(taskInstance));
+
+			if(planInstance.duration == null) planInstance.duration = [];
+			planInstance.duration.add(DateSpan(from: TimeDate.now()));
+			wasPlanInstanceChanged = true;
+		}
+		if(wasPlanInstanceChanged) toUpdate.add(planInstance);
+		updates.add(_dataRepository.updateMultiplePlanInstances(toUpdate));
+		await Future.value(updates);
+
+		Task task = await _dataRepository.getTask(taskId: taskInstance.taskID);
+		UITaskInstance uiTaskInstance = UITaskInstance.singleFromDBModel(task: taskInstance, name: task.name, description: task.description, points: task.points != null ? UIPoints(quantity: task.points.quantity, title: task.points.name, createdBy: task.points.createdBy, type: task.points.icon) : null);
+		_activeTaskService.setTaskStateActive();
+		if(isInProgress(uiTaskInstance.duration)) emit(TaskInstanceStateProgress(uiTaskInstance,  await _getUiPlanInstance(taskInstance.planInstanceID)));
+		else  emit(TaskInstanceStateBreak(uiTaskInstance,  await _getUiPlanInstance(taskInstance.planInstanceID)));
+	}
+
+	void switchToBreak() async {
+		TaskInstance taskInstance = await _dataRepository.getTaskInstance(taskInstanceId: _taskInstanceId);
+		taskInstance.duration.last.to = TimeDate.now();
+		taskInstance.breaks.add(DateSpan(from: TimeDate.now()));
+		await _dataRepository.updateTaskInstance(taskInstance);
+		Task task = await _dataRepository.getTask(taskId: taskInstance.taskID);
```

Comment:
You are also fetching the task in each function but it should be the same each time, why not do it once in constructor?